pkg/controller/secretgenerator/secretgenerator_controller.go

```diff
@@ -138,3 +154,52 @@ func (r *ReconcileSecretGenerator) Reconcile(request reconcile.Request) (reconci
 	// Don't requeue if secret is created successfully
 	return reconcile.Result{}, err
 }
+
+func (r *ReconcileSecretGenerator) createOAuthClient(foundSecret *v1.Secret, secret *Secret) error {
```

Comment:
We can split this function into two more generic functions to improve reusability and maintainability:

```golang
// getRoute returns an OpenShift route object. It waits until the .spec.host value exists to avoid possible race conditions, fails otherwise.  
func (r *ReconcileSecretGenerator) getRoute(name string, namespace string) (*routev1.Route, error) {
   // return routev1.Route{}, nil
}

// createOAuthClient creates an OpenShift OAuthClient object from the input secret and uri data.
func (r *ReconcileSecretGenerator) createOAuthClient(name string,
	namespace string, secret string, uri string, owner metav1.OwnerReference) error {
  // return nil
}
```

In general, all the `log.Infof` and `log.Errorf` should be handled in the `Reconcile` function. These functions should only return the error, not print it.